### ISSUE TITLE
Unable to create first account

### ISSUE THREAD

**Issue by kmccurley** [kohler/hotcrp#283]
I just cloned the code last night from kohler/hotcrp and created a first account. I received the email link, but when I went to the link it says:
![image](https://user-images.githubusercontent.com/14105363/186836891-e71d4e32-d229-411f-bb83-eaa99b6a0029.png)
I can see that the account exists in contactInfo with 
```
contactId: 1
password: 'unset'
passwordTime:1661495593
defaultWatch: 2
roles: 2
cdbRoles: 2
disabled: 0
primaryContactId: 0
```
This all looks normal, but the code in src/pages/p_signing.php seems to be generating this error code. I'm at a loss to figure out what is going on, but I see that there was a commit https://github.com/kohler/hotcrp/commit/7c12bc91a68ddb6817e18848c860e2320238c94d that appears to change some things in conf->cached_user_by_email() to conf->user_by_email(). I have no smoking gun but I've never seen this behavior before in any version of hotcrp.

**Comment by kmccurley** [comment]
I repeated it today, pulling a fresh install from master. As soon as the person creates the first account, the first thing the user sees is this. Clicking on the link in the email to the first user shows the same thing.
![image](https://user-images.githubusercontent.com/14105363/187094680-308ef1bb-d662-4965-b1ba-65a5ae3172ad.png)

**Comment by kohler** [comment]
Thank you for reporting this problem. Several changes over the last months have implicitly assumed a “contact database”, which is a shared user database not including per-conference data. Most installations do not include a contact database, but the test suite did. I updated the tests to (1) check the first-user workflow, (2) check most functionality both with & without a contact database. I've made a new release including those fixes. If you still have problems with it lmk here.
